src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -1238,7 +1242,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_mpi_accumulate(const void *origin_addr,
            !(winattr & MPIDI_WINATTR_ACCU_NO_SHM) ||
 #endif
            !MPIDI_OFI_ENABLE_RMA || !MPIDI_OFI_ENABLE_ATOMICS ||
-           !(winattr & MPIDI_WINATTR_NM_REACHABLE)) {
+           !MPIDI_OFI_is_reachable_target(winattr, origin_addr, NULL, NULL)) {
         mpi_errno =
```

Comment:
Likely we will do the same check in `cmpl_hook` in ofi_win.h similar to that in UCX. This allows us to avoid unnecessary network flush if all RMAs are AM based. Thus, it might be cleaner if we separate the target-side check (i.e., `(winattr & MPIDI_WINATTR_NM_REACHABLE)`)  and the origin-side check (i.e., all local buffers).